docs/src/compatibility.md

```diff
@@ -104,3 +104,20 @@ PkgB = "≥ 1.2.3"  # [1.2.3,  ∞)
 PkgC = "= 1.2.3"  # [1.2.3, 1.2.3]
 PkgD = "< 1.2.3"  # [0.0.0, 1.2.2]
 ```
+
+### Hyphen specifiers
+
```

Comment:
Did we decide to backport this? Maybe we can add something like this for now and update later when we backport?

```suggestion

!!! compat "Julia 1.4"
    Hyphen specifiers requires at least Julia 1.4, so it is recomended to also add
    ```toml
    [compat]
    julia = "1.4"
    ```
    to the project file when using them.
```

Reply:
Is package's Project.toml loaded in normal installation operations (i.e., `Pkg.add("Package"); using Package`)?  If not, isn't requiring `julia = "1.4"` overly restricting?  Though I suppose it can be problematic if you want to use `Pkg.test` to run tests (but there are other ways to run tests).

Reply:
> Is package's Project.toml loaded in normal installation operations (i.e., Pkg.add("Package"); using Package)? 

No, it isn't.

>  If not, isn't requiring julia = "1.4" overly restricting? Though I suppose it can be problematic if you want to use Pkg.test to run tests (but there are other ways to run tests).

Yeah, it's a bit odd situation that you can add the package but you cannot test it or dev it etc.